typescript/src/tomas_toniut/StringCalculator.ts

```diff
@@ -5,9 +5,13 @@ export class StringCalculator {
     if (input === '')
       return 0;
     const numbers = input.split(/[, \n]+/);
-    console.log(numbers)
     //el reduce toma la variable 'sum' como acumuladr y 'num' como el valor actual de la iteración. El cero al final es el valor inicial de acumulador. 
-    const result = numbers.reduce((sum, num) => sum + parseInt(num), 0)
+    const result = numbers.reduce((sum, num) => {
+      const parsedNum = parseInt(num);
+      if (parsedNum < 0) throw new Error(`negativos no están permitidos: ${numbers.join(',')}`);
```

Comment:
Entiendo que este código no va a pasar la especificación original:

> Especificación: `-1,2,-3` debe lanzar excepción `negatives not allowed: -1,-3`

Resultado actual para `-1,2,-3` será `negatives not allowed: -1,2,-3`

¿Se me escapa algo?